### PR TITLE
Add hamishknight to various directories in CODEOWNERS

### PR DESCRIPTION
**Pull request by bnbarham**
The primary change here is to add @hamishknight to directories he was missing, but while here:
  - Use alphabetical order for changed files/directories
  - Add all @ahoppen / @bnbarham / @hamishknight / @rintaro for any area
    that one of us is an owner of and remove @zoecarver
  - Add @ahoppen / @bnbarham / @hamishknight / @rintaro to ASTGen

### CODE REVIEW

**Comment by bnbarham** [comment]
@CodaFi / @zoecarver I'm happy to keep you in if you want to be notified about commits to these areas, up to you.

**Comment by bnbarham** [comment]
@swift-ci please smoke test

**Review by ahoppen — APPROVED** [review]

**Comment by zoecarver** [comment]
You can remove me. Thanks Ben.

**Comment by CodaFi** [comment]
I enjoy seeing the progress getting made here, and I think I still would like to be involved. Just don't expect a timely stamp is all I ask!

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -105,18 +105,18 @@
...
-/lib/IDE/                                           @ahoppen @bnbarham @rintaro @hamishknight
-/lib/IDETool/                                       @ahoppen @bnbarham @rintaro @hamishknight
-/lib/Index/                                         @bnbarham
-/lib/Refactoring/                                   @ahoppen @bnbarham
-/lib/IRGen/*Coverage*                               @hamishknight
+/lib/IDE/                                           @ahoppen @bnbarham @hamishknight @rintaro
+/lib/IDETool/                                       @ahoppen @bnbarham @hamishknight @rintaro
+/lib/Index/                                         @ahoppen @bnbarham @hamishknight @rintaro
+/lib/Refactoring/                                   @ahoppen @bnbarham @hamishknight @rintaro
+/lib/IRGen/*Coverage*                               @ahoppen @bnbarham @hamishknight @rintaro
```
Comment (AnthonyLatsis) [inline .github/CODEOWNERS:112]: May I take the opportunity to ask you to fix the order of these patterns? `IRGen` should go before `Index` and `Refactoring` should be moved further down.

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -163,13 +163,13 @@
...
-/test/SourceKit/                                    @ahoppen @bnbarham @rintaro @hamishknight
+/test/IDE/                                          @ahoppen @bnbarham @hamishknight @rintaro
+/test/Index/                                        @ahoppen @bnbarham @hamishknight @rintaro
+/test/refactoring/                                  @ahoppen @bnbarham @hamishknight @rintaro
+/test/SourceKit/                                    @ahoppen @bnbarham @hamishknight @rintaro
 /test/Interop/                                      @zoecarver @hyp @egorzhdan
-/test/Parse/                                        @ahoppen @bnbarham @CodaFi @DougGregor @rintaro
-/test/Profiler                                      @hamishknight
+/test/Parse/                                        @ahoppen @bnbarham @CodaFi @DougGregor @hamishknight @rintaro
+/test/Profiler                                      @ahoppen @bnbarham @hamishknight @rintaro
```
Comment (AnthonyLatsis) [inline .github/CODEOWNERS:172]: Same here.

**Review by AnthonyLatsis — APPROVED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (bnbarham) [reply .github/CODEOWNERS]: Sure, done 👍

**Comment by bnbarham** [comment]
@swift-ci please smoke test Linux platform

**Comment by bnbarham** [comment]
@swift-ci please smoke test Linux platform

**Comment by bnbarham** [comment]
@swift-ci please smoke test Linux platform

**Review by xedin — APPROVED** [review]

**Pull request merged**